File: frontend/src/types/CustomField.ts
import { CustomFieldType } from '../constants/Enumeration/CustomFieldEnum/CustomFieldEnum';

export type TCustomField = {
  id: string;
  workspaceId: string;
  fieldName: string;
  fieldType: CustomFieldType;
  defaultValue?: string;
  dropdownOptions?: string[];
  isRequired: boolean;
  isArchived: boolean;
  createdAt: string;
  updatedAt: string;
};

export type TAddCustomField = {
  workspaceId: string;
  fieldName: string;
  fieldType: CustomFieldType;
  defaultValue?: string;
  dropdownOptions?: string[];
  isRequired: boolean;
};

export type TUpdateCustomField = {
  id: string;
  fieldName?: string;
  fieldType?: CustomFieldType;
  defaultValue?: string;
  dropdownOptions?: string[];
  isRequired?: boolean;
  isArchived?: boolean;
};

File: backend/Models/CustomField.cs
using System.ComponentModel.DataAnnotations;

namespace backend.Models;

public class CustomField
{
    [Key]
    public Guid Id { get; set; }
    public Guid WorkspaceId { get; set; }

    [Required]
    public string FieldName { get; set; } = string.Empty;

    [Required]
    public CustomFieldType FieldType { get; set; } = CustomFieldType.Text;

    public string? DefaultValue { get; set; }
    public List<string>? DropdownOptions { get; set; }

    public bool IsRequired { get; set; } = false;
    public bool IsArchived { get; set; } = false;


    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

public enum CustomFieldType
{
    Text,
    Number,
    Date,
    Dropdown,
    Checkbox
}
